src/settings/value_type.rs

```
@@ -95,6 +96,23 @@ impl ValueType {
                 // Test if valid address falls within permitted ranges
                 value.is_empty() || is_ipv4_valid(value)
             }
+            ValueType::Ipv4optPort => {
+                // Valid, in allowable range, with optional port
+                // (4 octets, with port from 0 to 65535, colon delimited), or null
+                if value.is_empty() || is_ipv4_valid(value) {
+                    // check for valid ipv4 (without port)
+                    return true;
+                }
+                if !value.contains(":") {
+                    return false;
+                }
+
+                // check if port is specified
+                let (ip, portnumber) = value.split_at(value.rfind(":").unwrap_or_default());
+                let port = portnumber.parse::<usize>().unwrap_or_default();
```

These `unwrap_or_default` calls allow invalid inputs by ignoring the error cases.